src/LocationPicker.tsx

```diff
@@ -180,6 +182,26 @@ export default class LocationPicker extends Component<
       map.leafletElement.setView(startPort.center, startPort.zoom);
     }
   };
+  private renderTileLayer = () => {
+    const { geoserver, geoLayer, geoPort, geoWorkspace } = this.props;
+    const tileLayer = geoserver
+      ? {
+          url: `http://localhost:${geoPort}/geoserver/${geoWorkspace}/wms?`,
+          layers: geoLayer,
+          attribution: "Geoserver"
+        }
+      : {
+          url: "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
+          layers: "",
+          attribution:
+            '&amp;copy <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+        };
+    if (geoserver) {
```

Comment:
You do this same check above when assigning tileLayer. You could simplify by only checking this once and doing all the logic in the related sections.